cypress/e2e/talks-interaction.cy.ts

```diff
@@ -0,0 +1,238 @@
+  });
+
+  it("Should mimic real user interactions with talks buttons and show all talks in each category", () => {
+    cy.get("@git-pl").should("be.visible");
+    cy.get("@steem-pl").should("be.visible");
+    cy.get("@bitcoins-en").should("be.visible");
+    cy.get("@blockchain2-pl").should("be.visible");
+    cy.get("@blockchain1-pl").should("be.visible");
+    cy.get("@lightning-en").should("be.visible");
+    cy.get("@docker-pl").should("be.visible");
+
+    cy.get('[data-filter="git"]').click();
+    cy.get('#talk-filters > [data-filter="*"]').should(
+      "not.have.class",
+      "active"
+    );
+    cy.get('[data-filter="git"]').should("have.class", "active");
+    cy.get('[data-filter="docker"]').should("not.have.class", "active");
+    cy.get('[data-filter="blockchain"]').should("not.have.class", "active");
+    cy.get('#language-filters > [data-filter="*"]').should(
+      "have.class",
+      "active"
+    );
+    cy.get("@steem-pl").should("not.be.visible");
+    cy.get("@bitcoins-en").should("not.be.visible");
+    cy.get("@blockchain2-pl").should("not.be.visible");
+    cy.get("@blockchain1-pl").should("not.be.visible");
+    cy.get("@lightning-en").should("not.be.visible");
+    cy.get("@docker-pl").should("not.be.visible");
+
+    cy.get('[data-filter="docker"]').click();
+    cy.get('#talk-filters > [data-filter="*"]').should(
+      "not.have.class",
+      "active"
+    );
+    cy.get('[data-filter="git"]').should("not.have.class", "active");
+    cy.get('[data-filter="docker"]').should("have.class", "active");
+    cy.get('[data-filter="blockchain"]').should("not.have.class", "active");
+    cy.get('#language-filters > [data-filter="*"]').should(
+      "have.class",
+      "active"
+    );
+    cy.get("@git-pl").should("not.be.visible");
+    cy.get("@steem-pl").should("not.be.visible");
+    cy.get("@bitcoins-en").should("not.be.visible");
+    cy.get("@blockchain2-pl").should("not.be.visible");
+    cy.get("@blockchain1-pl").should("not.be.visible");
+    cy.get("@lightning-en").should("not.be.visible");
+    cy.get("@docker-pl").should("be.visible");
+
+    cy.get('[data-filter="blockchain"]').click();
+    cy.get('#talk-filters > [data-filter="*"]').should(
+      "not.have.class",
+      "active"
+    );
+    cy.get('[data-filter="git"]').should("not.have.class", "active");
+    cy.get('[data-filter="docker"]').should("not.have.class", "active");
+    cy.get('[data-filter="blockchain"]').should("have.class", "active");
+    cy.get('#language-filters > [data-filter="*"]').should(
+      "have.class",
+      "active"
+    );
+    cy.get("@git-pl").should("not.be.visible");
+    cy.get("@steem-pl").should("be.visible");
+    cy.get("@bitcoins-en").should("be.visible");
+    cy.get("@blockchain2-pl").should("be.visible");
+    cy.get("@blockchain1-pl").should("be.visible");
+    cy.get("@lightning-en").should("be.visible");
+    cy.get("@docker-pl").should("not.be.visible");
+  });
```

Comment:
We could create a function `assertFiltersWorksCorrectly(itemToSelect, expectedListOfItemsAvailable)`.